org.eclipse.lsp4e/src/org/eclipse/lsp4e/LanguageServerWrapper.java

```diff
@@ -713,13 +756,7 @@ protected CompletableFuture<LanguageServer> getInitializedServer() {
 
 		if (initializeFuture != null && !this.initializeFuture.isDone()) {
 			if (Display.getCurrent() != null) { // UI Thread
-				final var waitForInitialization = new Job(Messages.initializeLanguageServer_job) {
-					@Override
-					protected IStatus run(IProgressMonitor monitor) {
-						initializeFuture.join();
-						return Status.OK_STATUS;
-					}
-				};
+				final Job waitForInitialization = createInitializeLanguageServerJob();
```

Comment:
@joaodinissf , is initializeFutureNumberOfStages = 7 correct also for this case?

Reply:
I think it can happen that the number of stages is not accurate for the UI pop-up, but since I think we should remove the pop-up in a new PR, we can merge this one nevertheless.

Reply:
I'll create a separate PR for the removal of the pop-up.